### ISSUE TITLE
In JS, imported class type references resolve to static, not instance type.

### ISSUE THREAD

**Issue by sandersn** [microsoft/TypeScript#34671]
```js
// @filename: first.js
class C {
  s() { }
}
module.exports.C = C
// @filename: test.js
/** @typedef {import('./first').C} C */
/** @param {C} c */
function demo(c) {
  c.s
}
```

Expected: no error and `c.s: () => {}`.

Actual: Property 's' does not exist on 'typeof C'.

**Comment by sandersn** [comment]
Broken by #34515
